### PR TITLE
Add image description support

### LINKED ISSUE #6860: Image description/caption
## Summary

Currently we can rename an image but not add a description or caption to it.

### LINKED ISSUE #7145: Add alt / description to uploaded media for better accessibility
In order to increase Misskey's accessibility it would be a greatly welcome addition to offer media description like Twitter or Mastodon do for visually impaired folks.

image

### PR DESCRIPTION
**Pull request by nullobsi**
## Summary
Resolve #7145
Resolve #6860

Adds support for both federation and addition of image descriptions using the ALT and TITLE properties on img.
Adds support to attach image descriptions to drive files and store them under comment in the database.

### CODE REVIEW

**Comment by syuilo** [comment]
Am I correct in understanding that in other systems (e.g. Mastodon), the description of the image is represented by the `name` property?

**Comment by nullobsi** [comment]
yes, Mastodon and Pleorma use name

**Comment by nullobsi** [comment]
I am not sure how to fix the test for 14.x or why it differs from 16.x...

File: locales/ja-JP.yml [hunk 1]
```diff
@@ -546,6 +547,8 @@ disablePlayer: "プレイヤーを閉じる"
 expandTweet: "ツイートを展開する"
 themeEditor: "テーマエディター"
 description: "説明"
+describeFile: "キャプションを付く"
```
Comment (rinsuki) [inline locales/ja-JP.yml]:
```suggestion
describeFile: "キャプションを付ける"
```

**Comment by nullobsi** [comment]
I have reused parts of the image-viewer component to make the captioning dialog a bit easier to use. I have built it on my instance, and it seems to be working...

**Comment by syuilo** [comment]
@nullobsi nice. So can you undo the changes in dialog.vue in this PR?

**Comment by nullobsi** [comment]
alright. The changes have been reverted

**Comment by syuilo** [comment]
👍👍👍

**Pull request merged**